app.js

```diff
@@ -147,6 +168,57 @@ router.post('/account/:phoneNumber/:accountId/validateCode', async ctx => {
     ctx.body = {};
 });
 
+function recoveryMethodsFor(account) {
+    return {
+        email: account.email || null,
+        emailAddedAt: account.emailAddedAt || null,
+        phoneAddedAt: account.phoneAddedAt || null,
+        phoneNumber: account.phoneNumber || null,
+        phraseAddedAt: account.phraseAddedAt || null,
+    };
+}
+
+router.post('/account/:accountId/recoveryMethods', async ctx => {
+    const { accountId } = ctx.params;
+    const { securityCode, signature } = ctx.request.body;
+
+    const account = await models.Account.findOne({ where: { accountId } });
+
+    await verifyAccountOwnership({ ctx, accountId, securityCode, signature });
+
+    ctx.body = recoveryMethodsFor(account || {});
+});
+
+const recoveryMethods = ['phone', 'email', 'phrase'];
+
+router.post('/account/deleteRecoveryMethod', async ctx => {
+    const { accountId, recoveryMethod, securityCode, signature } = ctx.request.body;
+
+    if (!recoveryMethods.includes(recoveryMethod)) {
```

Comment:
feel like it's more clear to have `switch (recoveryMethod)`?  with all the cases (including `default:` for this one)

Reply:
I've updated it with something I think you'll like more